dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Trn/CheckAnswers.cshtml.cs

```diff
@@ -31,8 +43,48 @@ public void OnGet()
     {
     }
 
-    public void OnPost()
+    public async Task<IActionResult> OnPost()
     {
+        var authenticationState = HttpContext.GetAuthenticationState();
+
+        var userId = Guid.NewGuid();
+        var user = new User()
+        {
+            CompletedTrnLookup = _clock.UtcNow,
+            Created = _clock.UtcNow,
+            DateOfBirth = authenticationState.DateOfBirth,
+            EmailAddress = authenticationState.EmailAddress!,
+            FirstName = string.IsNullOrEmpty(authenticationState.FirstName) ? authenticationState.OfficialFirstName! : authenticationState.FirstName,
+            LastName = string.IsNullOrEmpty(authenticationState.LastName) ? authenticationState.OfficialLastName! : authenticationState.LastName,
+            Updated = _clock.UtcNow,
+            UserId = userId,
+            UserType = UserType.Default,
+            Trn = authenticationState.StatedTrn,
+            TrnAssociationSource = !string.IsNullOrEmpty(authenticationState.Trn) ? TrnAssociationSource.Lookup : null,
+            LastSignedIn = _clock.UtcNow,
+            RegisteredWithClientId = authenticationState.OAuthState?.ClientId,
+            TrnLookupStatus = !string.IsNullOrEmpty(authenticationState.Trn) ? TrnLookupStatus.Found : TrnLookupStatus.None
+        };
+
+        _dbContext.Users.Add(user);
+
+        _dbContext.AddEvent(new Events.UserRegisteredEvent()
+        {
+            ClientId = authenticationState.OAuthState?.ClientId,
+            CreatedUtc = _clock.UtcNow,
+            User = user
+        });
+
+        try
+        {
+            await _dbContext.SaveChangesAsync();
+        }
+        catch (DbUpdateException dex) when (dex.IsUniqueIndexViolation("ix_users_email_address"))
+        {
+            return Redirect(_linkGenerator.TrnInUseChooseEmail());
```

Comment:
Not doing any exception handling here except redirecting to TrnInUseChooseEmail

Reply:
Need to replicate what's happening in `TrnCallback` here (find the user with the existing email, generate PIN etc.)